keylime/secure_mount.py

```diff
@@ -20,7 +20,7 @@
 def check_mounted(secdir):
     """Inspect mountinfo to detect if a directory is mounted."""
     secdir_escaped = secdir.replace(" ", r"\040")
-    for line in open("/proc/self/mountinfo", "r", encoding="utf-8"):
+    for line in open("/proc/self/mountinfo", "r", encoding="utf-8"):  #pylint: disable=consider-using-with
```

Comment:
Can we not open the file with `with` and then iterate over `readline()`?

Reply:
This seems to work fine for me with python 3.10 just running this here:

```
>>> with open("/proc/self/mountinfo", "r", encoding="utf-8") as f:
...     for line in f:
...         print(line)
```

On the CI/CD system we now get this here:

```
Error in test.test_secure_mount.TestSecureMount.test_check_mounted_found
  File "/usr/lib64/python3.9/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib64/python3.9/unittest/case.py", line 592, in run
    self._callTestMethod(testMethod)
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/usr/lib64/python3.9/unittest/mock.py", line 1336, in patched
    return func(*newargs, **newkeywargs)
  File "/__w/keylime/keylime/test/test_secure_mount.py", line 31, in test_check_mounted_found
    self.assertTrue(secure_mount.check_mounted("/secdir"))
  File "/__w/keylime/keylime/keylime/secure_mount.py", line 23, in check_mounted
    with open("/proc/self/mountinfo", "r", encoding="utf-8") as f:
AttributeError: __enter__
```

Reply:
This is beacuse the open call is mocked, but the `__enter__` and `__exit__` functions are not: https://github.com/keylime/keylime/blob/master/test/test_secure_mount.py

Reply:
So odd...
```
>>> subprocess.Popen.__enter__
<function Popen.__enter__ at 0x6fffffdf2d30>
>>> open.__enter__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'builtin_function_or_method' object has no attribute '__enter__'
>>> os.open.__enter__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'builtin_function_or_method' object has no attribute '__enter__'
```

Reply:
There seems to be a mocking object in the library for that: https://docs.python.org/3.7/library/unittest.mock.html#mock-open

Reply:
Fixed in additional patch. What a struggle!